File: LeetCode Algorithms/code/11.盛最多水的容器.py
#
# @lc app=leetcode.cn id=11 lang=python3
#
# [11] 盛最多水的容器
#

# @lc code=start
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left_index, right_index, res = 0, len(height) - 1, 0
        while left_index < right_index:
            if height[left_index] < height[right_index]:
                res = max(res, height[left_index] * (right_index - left_index))
                left_index += 1
            else:
                res = max(res, height[right_index] * (right_index - left_index))
                right_index -= 1
        return res
# @lc code=end

File: LeetCode Algorithms/11.盛最多水的容器.cpp
/*
 * @lc app=leetcode.cn id=11 lang=cpp
 *
 * [11] 盛最多水的容器
 */

// @lc code=start
class Solution {
public:
    int maxArea(vector<int>& height) {
        int max_res = 0, left_index =0, right_index = height.size();
        while (left_index <= right_index) {
            if (height[left_index] < height[right_index]) {
                max_res = max(max_res, height[left_index] * (right_index - left_index));
                left_index ++;
            }
            else {
                max_res = max(max_res, height[right_index] * (right_index - left_index));
                right_index --;
            }
        }
        return max_res;
    }
};
// @lc code=end

